File: _50_Pow(x,n).py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        result = 1
        N = abs(n)

        while N:
            if N % 2:
                result *= x
            x *= x
            N //= 2

        if n > 0:
            return result
        else:
            return 1 / result

File: maths/_50_pow(x,_n).java
class Solution {
    public double myPow(double x, int n) {
        long binForm = n; 
        double result = 1; 

        if (binForm < 0) {
            x = 1 / x;
            binForm = -binForm;
        }

        while (binForm > 0) {
            if (binForm % 2 == 1) { // yahan last binary didgit check kar rahe hain
                result *= x;
            }

            x *= x;
            binForm /= 2; // yahan right binary digit ko remove kar rahe hain

        }

        return result;
    }
}